src/components/inspectors/TimerExpression.vue

```diff
@@ -0,0 +1,381 @@
+<template>
+  <div class="form-group">
+    <label>Start Date</label>
+    <div>
+      <datepicker v-model="startDate"
+                  calendar-class="calendar" format="yyyy-MM-dd"
+                  input-class="form-control start-date" 
+                  class="start-date-div"
+                  @selected="updateStartDate"></datepicker>
+      <select v-model="startTime" class="form-control control time" @change="update">
+        <option v-for="hour in hours" :key="hour" :value="hour">{{hour}}</option>
+      </select>
+    </div>
+    <template v-if="hasRepeat">
+    <label>{{repeatLabel}}</label>
+    <div>
+      <input type="number" min="1" class="form-control control repeat" v-model="repeat" @change="update">
+      <select v-model="periodicity" class="form-control control periodicity" @change="update">
+        <option value="day">day</option>
+        <option value="week">week</option>
+        <option value="month">month</option>
+        <option value="year">year</option>
+      </select>
+    </div>
+    </template>
+    <div v-if="periodicity==='week'">
+      <label>{{weekLabel}}</label>
+      <div>
+        <span v-for="(day, index) in weekdays" :key="index + 'week'" 
+              class="badge badge-pill weekday"
+              :class="{'badge-primary': day.selected && !sameDay, 'badge-light': !(day.selected && !sameDay), 'border border-primary': currentDaySelected(day)}"
```

Comment:
Instead of having the conditions in the template, could it be a computed function, and just call the function which handles the logic?

Reply:
moved to **weekdayStyle**